sdk/forms/README.md

```diff
@@ -10,130 +10,276 @@
 
-```html
-<form action="/cards" method="post">
-  <h1>Add payment details</h1>
-  <label for="card_number">Card Number</label>
-  <input type="text" id="card_number" name="card_number" />
-  <label for="card_holder">Card Holder</label>
-  <input type="text" id="card_holder" name="card_holder" />
-  <label for="card_expiry">Card Expiry</label>
-  <input type="text" id="card_expiry" name="card_expiry" />
-  <label for="card_cvv">Card CVV</label>
-  <input type="text" id="card_cvv" name="card_cvv" />
-  <button type="submit">Add card</button>
-</form>
+## Table of Contents
+
+- [When to Use Piiano Forms](#when-to-use-piiano-forms)
+  - [Example Use Cases](#example-use-cases)
+- [How It Works](#how-it-works)
+- [Setup](#setup)
+- [API](#api)
+  - [createProtectedForm](#createprotectedform)
+    - [Overview](#overview)
+    - [Goal](#goal)
+    - [Usage](#usage)
+  - [createProtectedView](#createprotectedview)
+    - [Overview](#overview-1)
+    - [Goal](#goal-1)
+    - [Usage](#usage-1)
+  - [controlForm](#controlform)
+    - [Overview](#overview-2)
+    - [Goal](#goal-2)
+    - [Usage](#usage-2)
+- [Recommended Vault Authorization Techniques](#recommended-vault-authorization-techniques)
+- [Benefits of Using Piiano Vault with Piiano Forms](#benefits-of-using-piiano-vault-with-piiano-forms)
+- [Getting Started](#getting-started)
+
+## When to Use Piiano Forms
+
+Piiano Forms is ideal for any use case where you want to collect sensitive information (e.g., credit card details, personal contact information) securely and make sure that it does not reach your backend in raw form.
+
+### Example Use Cases:
+
+- **Credit Card Payments**: Save credit card details in Vault and use Vault to pass them to payment gateways when
+  needed, without making your entire backend PCI compliant.
+- **Marketing Automation**: Save personal contact information like phone numbers and emails in Vault, and use Vault to
+  send it to marketing tools for sending emails and SMS, without exposing this data to your backend.
+- **Shipping Addresses**: Store addresses in Vault, and use it to pass them to shipping providers for deliveries,
+  maintaining complete privacy.
+
+## How It Works
+
+Piiano Forms interacts exclusively with Piiano Vault, ensuring that all sensitive data is securely tokenized, encrypted, or stored before any further processing. This interaction happens directly between the form and Piiano Vault, without involving your backend. This approach is beneficial as:
+
+1. The Vault can be configured to allow communication only from the allowed origin of the form.
+2. The form iframe is served with strict CSP (Content Security Policy) headers, making it a sandbox that prevents PII from leaking out, while allowing exclusive communication with the Vault.
+
+This architecture ensures that even if the user frontend is compromised (e.g., via a malicious package or XSS attack), the attacker cannot access the iframe content or manipulate the communication with the Vault.
```

Comment:
"the user frontend" --> you mean the hosting frontend. The term user would be confusing here